crates/bevy_ecs/src/name.rs

```diff
@@ -141,6 +147,36 @@ pub struct NameOrEntity {
     pub entity: Entity,
 }
 
+impl NameOrEntity {
+    /// Get a [`NameOrEntity`] from a [`Query`].
+    /// This is a convenience wrapper around [`Query::get`],
+    /// which uses the fact that a [`NameOrEntity`] can always be obtained,
+    /// even if the entity does not have a [`Name`].
```

Comment:
This line isn't quite right. The problem is not whether the entity has a name, but whether the entity is still "alive" (not despawned). Getting an entity without a name works already!

Reply:
Ah right, probably messed that up in my "quick writing" xD, will fix this!